tests/unit/test_charm.py

```diff
@@ -1406,12 +1420,70 @@ def test_ingresses_for_multiple_relations_different_hostnames(
         mock_create_ingress.assert_not_called()
         mock_replace_ingress.assert_not_called()
 
+    @patch("charm.NginxIngressCharm._report_service_ips")
+    @patch("charm.NginxIngressCharm.k8s_auth")
+    @patch("charm.NginxIngressCharm._define_ingress")
+    @patch("charm.NginxIngressCharm._remove_service")
+    @patch("charm.NginxIngressCharm._define_service")
+    @patch("charm._networking_v1_api")
+    def test_report_ingress_ips(
+        self,
+        mock_api,
+        mock_define_service,
+        mock_remove_service,
+        mock_define_ingress,
+        mock_k8s_auth,
+        mock_service_ips,
+    ):
+        """Test for checking the API call to obtain ingress IPs."""
+
+        mock_ingress = MagicMock()
+        mock_ingress.status.load_balancer.ingress[0].ip = "127.0.0.1"
+        mock_list_ingress = mock_api.return_value.list_namespaced_ingress
+        mock_list_ingress.return_value.items = [mock_ingress]
+
+        expected_result = ["127.0.0.1"]
+
+        result = NginxIngressCharm._report_ingress_ips(NginxIngressCharm)
+
+        self.assertEqual(result, expected_result)
+
+    @patch("charm.NginxIngressCharm._report_service_ips")
+    @patch("charm.NginxIngressCharm.k8s_auth")
+    @patch("charm.NginxIngressCharm._define_ingress")
+    @patch("charm.NginxIngressCharm._remove_service")
+    @patch("charm.NginxIngressCharm._define_service")
+    @patch("charm._networking_v1_api")
+    def test_report_ingress_ips_fail(
+        self,
+        mock_api,
+        mock_define_service,
+        mock_remove_service,
+        mock_define_ingress,
+        mock_k8s_auth,
+        mock_service_ips,
+    ):
+        """
+        Test for checking the API call to obtain ingress IPs
```

Comment:
We have a standard for writing the docstring for tests, do you think it applies here? https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#test-structure